### PR TITLE
feat(server): runtime setting support

### PR DESCRIPTION
**Pull request by forehalo**
---

<details open="true"><summary>Generated summary (powered by <a href="https://app.graphite.dev">Graphite</a>)</summary>

> ## TL;DR
> This pull request adds a new migration file, a new model, and new modules related to runtime settings. It also introduces a new `Runtime` service that allows getting, setting, and updating runtime configurations.
> 
> ## What changed
> - Added a new migration file `migration.sql` that creates a table called `application_settings` with columns `key` and `value`.
> - Added a new model `ApplicationSetting` with properties `key` and `value`.
> - Added a new module `RuntimeSettingModule` that exports the `Runtime` service.
> - Added a new service `Runtime` that provides methods for getting, setting, and updating runtime configurations.
> - Modified the `app.module.ts` file to import the `RuntimeSettingModule`.
> - Modified the `index.ts` file in the `fundamentals` directory to export the `Runtime` service.
> - Added a new file `def.ts` in the `runtime` directory that defines the runtime configurations and provides a default implementation.
> - Added a new file `service.ts` in the `runtime` directory that implements the `Runtime` service.
> 
> ## How to test
> 1. Run the migration script to create the `application_settings` table.
> 2. Use the `Runtime` service to get, set, and update runtime configurations.
> 3. Verify that the runtime configurations are stored correctly in the database and can be retrieved and modified using the `Runtime` service.
> 
> ## Why make this change
> This change introduces a new feature related to runtime settings. The `Runtime` service allows the application to dynamically manage and modify runtime configurations without requiring a restart. This provides flexibility and allows for easier customization and configuration of the application.
</details>

### CODE REVIEW

**Comment by forehalo** [comment]
* **#7092** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7092?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7082** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7082?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5602** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5602?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6995** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6995?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @forehalo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/src/core/config/resolver.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+  @Query(() => [ServerRuntimeConfigType], {
+    description: 'get all server runtime configurable settings',
+  })
+  serverRuntimeConfig(): Promise<ServerRuntimeConfigType[]> {
+    // TODO: permission check
+    return this.config.runtime.list();
+  }
+
+  @Mutation(() => ServerRuntimeConfigType, {
```
Comment (Brooooooklyn) [inline packages/backend/server/src/core/config/resolver.ts:183]: permission guard?

Comment (forehalo) [reply packages/backend/server/src/core/config/resolver.ts:183]: in next pr

**Review by Brooooooklyn — APPROVED** [review]

File: packages/backend/server/src/fundamentals/config/default.ts [hunk 2]
```diff
@@ -83,124 +44,85 @@ export const getDefaultAFFiNEConfig: () => AFFiNEConfig = () => {
...
+  const result = { ...a };
+  Object.keys(b).forEach(key => {
+    result[key] = merge(result[key], b[key]);
+  });
+
+  return result;
+}
+
+export function mergeConfigOverride(override: any) {
+  return merge(defaultStartupConfig, override);
```
Comment (darkskygit) [inline packages/backend/server/src/fundamentals/config/default.ts:95]: why not lodash-es.merge?

Comment (forehalo) [reply packages/backend/server/src/fundamentals/config/default.ts:95]: it's buggy when merging proxied objects

**Review by EYHN — APPROVED** [review]

**Review by darkskygit — APPROVED** [review]

**Pull request merged**